web/static/js/main.js

```diff
@@ -68,29 +68,33 @@ function create_top_values_to_plot_structure(event_type, element){
 
 /**
  * Get top 10 element values in the given event type and plot them.
- * @param {*} event_type 
- * @param {*} element 
- * @param {*} html_element_id 
+ * @param {*} event_type
+ * @param {*} element
+ * @param {*} html_element_id
  */
 function get_top_ten_element_in_event(event_type, element, html_element_id){
     $.ajax({
         type: "GET",
-        url: "/api/events/count/groupby/"+event_type.toLowerCase()+"/"+element.toLowerCase(),
+        url: "/api/events/count/groupby/" + event_type.toLowerCase() + "/" + element.toLowerCase(),
         success: function(result,status,xhr){
             create_top_values_to_plot_structure(event_type, element);
-            for (var i = 0; i < result.length; i++) {
-                top_values_to_plot[event_type][element].keys.push(
-                    result[i][Object.keys(result[i])[1]]
-                );
-                top_values_to_plot[event_type][element].values.push(result[i][Object.keys(result[i])[0]]);
-                top_values_to_plot[event_type][element].colors.push(color(colors_array[i]).alpha(0.5).rgbString());
+            const keys = Object.keys(result);
+            for (let i = 0; i < keys.length; i++) {
+                const index = top_values_to_plot[event_type][element].keys.indexOf(keys[i]);
+                if (index === -1) {
```

Comment:
i think it's not needed to check if it exists this ways because create_top_values_to_plot_structure function does that already. 
If I misunderstood this, you can correct me by giving an example